fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetaClient.java

```diff
@@ -403,6 +388,86 @@ private List<Partition> getPartitionsWithRetry(String dbName, String tableName,
+        try (Timer ignored = Tracers.watchScope(EXTERNAL, "HMS.getPartitionColumnStats")) {
+            RecyclableClient client = null;
+            StarRocksConnectorException connectionException = null;
+            try {
+                client = getClient();
+                partitionStats = client.hiveClient.getPartitionColumnStatistics(dbName, tblName, partitionNames, columnNames);
+                if (partitionStats.size() != partitionNames.size()) {
+                    LOG.warn("Expect to fetch {} partitionStats on [{}.{}], but actually fetched {} partition",
+                            partitionNames.size(), dbName, tblName, partitionStats.size());
+                }
+            } catch (TTransportException te) {
+                partitionStats = getPartitionColumnStatsWithRetry(dbName, tblName, partitionNames, columnNames, 1);
+            } catch (Exception e) {
+                LOG.error("Failed to get partitionStats on {}.{}", dbName, tblName, e);
+                connectionException = new StarRocksConnectorException("Failed to get partitionStats on [%s.%s]" +
+                        " from meta store: %s", dbName, tblName, e.getMessage());
+                throw connectionException;
+            } finally {
+                if (client == null && connectionException != null) {
+                    LOG.error("Failed to get hive client. {}", connectionException.getMessage());
+                } else if (connectionException != null) {
+                    LOG.error("An exception occurred when using the current long link " +
+                            "to access metastore. msg: {}", connectionException.getMessage());
+                    client.close();
+                } else if (client != null) {
+                    client.finish();
+                }
+            }
+        }
+        return partitionStats;
+    }
+
+    private Map<String, List<ColumnStatisticsObj>> getPartitionColumnStatsWithRetry(String dbName,
+                                                                                    String tableName,
+                                                                                    List<String> partNames,
+                                                                                    List<String> columnNames,
+                                                                                    int retryNum)
+            throws StarRocksConnectorException {
+        int subListSize = (int) Math.pow(2, retryNum);
+        int subListNum = partNames.size() / subListSize;
+        if (subListNum == 0) {
+            subListNum = 1;
+        }
+        List<List<String>> partNamesList = Lists.partition(partNames, subListNum);
+        Map<String, List<ColumnStatisticsObj>> partitionStats = new HashMap<>();
+
+        LOG.warn("Execute getPartitionColumnStatistics on [{}.{}] with {} times retry, slice size is {}, partName size is {}",
```

Comment:
This error message is not accurate, current meaning is `the nth retry`, but not the `with N times retry`

Reply:
Not too much?